vote/src/vote_account.rs

```diff
@@ -139,16 +138,20 @@ impl VoteAccounts {
     pub fn staked_nodes(&self) -> Arc<HashMap</*node_pubkey:*/ Pubkey, /*stake:*/ u64>> {
         self.staked_nodes
             .get_or_init(|| {
-                Arc::new(
-                    self.vote_accounts
-                        .values()
-                        .filter(|(stake, _)| *stake != 0u64)
-                        .map(|(stake, vote_account)| (*vote_account.node_pubkey(), stake))
-                        .into_grouping_map()
-                        .aggregate(|acc, _node_pubkey, stake| {
-                            Some(acc.unwrap_or_default() + stake)
-                        }),
-                )
+                // Pre-allocate HashMap with estimated capacity to reduce reallocations
+                let mut staked_nodes =
+                    HashMap::with_capacity(self.vote_accounts.len().saturating_div(2));
```

Comment:
Given that this is a hash map of validators, I think we could use `PubkeyHasherBuilder` instead of the default hasher:

```suggestion
                    HashMap::with_capacity_and_hasher(self.vote_accounts.len().saturating_div(2), PubkeyHasherBuilder::default());
```

That should speed it up even more. :slightly_smiling_face: You'll need to add the `PubkeyHasherBuilder` generic to the return type in the function signature.

Reply:
PubkeyHasherBuilder is behind "rand" feature. vote package doesn't enable rand feature. 
What's your experience on enable rand feature? Would it be a more broader change and require more testing to ensure no regression?
How about doing it as a separate optimization in a follow-up pr?